src/plugins/intel_gpu/src/graph/impls/ocl/roll.cpp

```diff
@@ -27,8 +27,43 @@ struct roll_impl : typed_primitive_impl_ocl<roll> {
         const auto& primitive = impl_param.typed_desc<roll>();
         auto params = get_default_params<kernel_selector::roll_params>(impl_param);
 
-        params.shift = convert_dim_vector(primitive->shift);
-
+        if ((primitive->raw_shift.empty()) && (primitive->raw_axes.empty())) {
```

Comment:
How does it work without adding new entry to implementation_map with dynamic shape support?
Could you please add related functional tests for dynamic scenario?

Reply:
@sshlyapn Oops! I missed to add the func test file in the commit! I added it now. Thanks!

Reply:
@wilson-seok thanks!
Could you please define `get_shape_infer_dependencies()` function for roll node? Just to avoid unnecessary synchronizations
BTW, is this PR only about functional support? And shape agnostic kernel would be added later if needed?

Reply:
@sshlyapn Ok, I added get_shape_infer_dependencies() function for roll node.
Yes, currently I added only dynamic shape capability of roll op first to fix several tickets. Will implement shape agnostic kernel later if needed.

Reply:
@wilson-seok  I see, thank you